src/sage/matrix/matrix_integer_dense.pyx

```diff
@@ -3250,6 +3281,18 @@ cdef class Matrix_integer_dense(Matrix_dense):
             R = A.to_matrix(self.new_matrix())
             if transformation:
                 U = U.to_matrix(self.new_matrix(U.nrows, U.ncols))
+
+        elif algorithm == 'pari':
+            # call pari with flag=4: kernel+image
+            # pari uses column convention: need to transpose the matrices
+            from sage.libs.pari import pari
+            A = pari(self).mattranspose()
+            K, T = A.qflll(4)
+            U = pari.matconcat([K, T]).mattranspose().sage()
+            R = U * self
+            r = ZZ(T.length())
```

Comment:
This is a pari method call done after lines where the sage language is performed. Maybe keep pari computations all together is more readable.